tests/foreman/cli/test_bootstrap_script.py

```diff
@@ -15,62 +15,205 @@
+            location=[cls.loc],
+            content_view=cv,
+            lifecycle_environment=lce.id,
+            organization=[cls.org],
+            operatingsystem=os,
+                ).create()
+
+    @tier4
     def test_positive_register(self):
-        """System is registered
+        """System is registered.
 
         :id: e34561fd-e0d6-4587-84eb-f86bd131aab1
 
         :Steps:
 
-            1. assure system is not registered
-            2. register a system
+            1. create a container with host name
+            2. register system using bootstrap.py
+            3. assert system has been registered
 
         :expectedresults: system is registered, host is created
 
-        :CaseAutomation: notautomated
+        :CaseAutomation: automated
 
-        :CaseImportance: Critical
+        :CaseImportance: High
         """
+        my_host = Container(agent=True)
+        host_name = my_host.execute("hostname")
+        my_fqdn = host_name.strip() + '.' + self.domain.name + '.' 'com'
+        my_host.execute("curl -O http://{}/pub/bootstrap.py".format(settings.server.hostname))
+        result = my_host.execute("python bootstrap.py -l admin -p changeme -s {} -o '{}' "
+                                 "-L '{}' -g {} -a {} --fqdn {} --force --add-domain"
+                                 .format(
+                                     settings.server.hostname,
+                                     self.org.name, self.loc.name,
+                                     self.hostgroup.name,
+                                     self.ak.name,
+                                     my_fqdn
+                                         )
+                                 )
+        assert "The system has been registered" in result, 'Not registered'
+        Container.delete(my_host)
 
     @tier1
-    @stubbed()
     @upgrade
     def test_positive_reregister(self):
-        """Registered system is re-registered
+        """Registered system is re-registered.
 
         :id: d8a7aef1-7522-47a8-8478-77e81ca236be
 
         :Steps:
 
-            1. register a system
+            1. register a system using commands
             2. assure system is registered
-            3. register system once again
+            3. register system once again using bootstrap.py
+            4. assure system is registered
 
         :expectedresults: system is newly registered, host is created
 
-        :CaseAutomation: notautomated
+        :BZ: 1739367
 
-        :CaseImportance: Critical
+        :CaseAutomation: automated
+
+        :CaseImportance: Medium
         """
+        my_host = Container(agent=True)
+        host_name = my_host.execute("hostname")
+        my_fqdn = host_name.strip() + '.' + self.domain.name + '.' 'com'
+        # Register host the first time
+        my_host.execute("curl -O http://{}/pub/bootstrap.py".format(settings.server.hostname))
+        result = my_host.execute("python bootstrap.py -l admin -p changeme -s {} -o '{}' "
```

Comment:
Do you actually need this `bootstrap.py` call?

Seems in bugzilla steps are:
1. run bootstrap.py
2. remove `katello-ca-consumer` and any package that could be used to report status back to host
3. install `katello-ca-consumer`
4. register host using `subscription-manager`
5. run bootstrap.py again

Could you start from step 3 above? Would it reproduce the issue as well?

Your steps are:
1. run bootstrap.py
2. register host using `subscription-manager`
3. run bootstrap.py again

Shouldn't you de-register and re-install `katello-ca-consumer` between steps 2 and 3, then?

Reply:
Hello @mirekdlugosz 

At first I did try register the first time with `subscription-manager` but I needed the `--add-domain` option the first time for some reason so I decided to just use bootstrap.py twice.

In `test_positive_reregister` I see:
1] run bootstrap.py first time on line 183
2] run `subscription-manager attach --auto` on line 193
3] run bootstrap.py second time on line 198

I will ask jturel about the need to de-register and re-install `katello-ca-consumer` between steps 2 and 3.

Thank you

Reply:
These test cases seem comprehensive since using bootstrap.py to register was broken completely in 6.5.2

I don't think there's any reason to manage the katello-ca-consumer RPM in any way - bootstrap.py does that on its own